src/pages/RequestListPage/RequestListPage.js

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import RequestList from '../../components/RequestList/RequestList';
+import MessagePrompt from '../../components/MessagePrompt/MessagePrompt';
+import { editRequest, setFormPrompt } from '../../app/redux';
+
+const RequestListPage = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [requests, setRequests] = useState([]);
+  const dispatch = useDispatch();
+  const msg = useSelector(state => state.form.msg);
+
+  useEffect(() => {
+    const getUser = async (id) => {
+      const uri = process.env.REACT_APP_BASE_API_URL + '/users?id=' + id;
+      const userResponse = await fetch(uri);
+      const user = await userResponse.json();
+      return user[0];
+    }
+
+    const getPendingRequests = async () => {
+      const uri = process.env.REACT_APP_BASE_API_URL + '/requests?status=pending';
+      const response = await fetch(uri);
+      const requestData = await response.json();
+
+      for(let i = 0; i < requestData.length; i++) {
+        requestData[i].user = await getUser(requestData[i].userId);
+      }
```

Comment:
Does this one make the flow slower if we have to wait for every request to be processed step by step?

Reply:
It does make the user wait for the action to finish, right now I am displaying a "loading..." message when the data is being fetched. To my knowledge this is the general practice when fetching data for a component. If there is any better way to improve the performance please let me know  😄